README.md

```diff
@@ -345,6 +345,33 @@ Examples
    # vmlinux available in the system.
 ```
 
+ - Connect to a simple remote shell:
+```
+   # Start the vng instance with vsock support:
+   $ vng --vsock
+
+   # In a separate terminal run the following command to connect to a remote shell:
+   $ vng --vsock-connect
+```
+
+ - Connect to a remote shell with proper dimensions, env vars, and using 'Fish':
+```
+   # Start the vng instance with vsock support:
+   $ vng --vsock "${PWD}/console.sh"
+
+   # In a separate terminal run the following commands:
+   $ read -r rows columns <<< "$(stty size)"
+   $ cat <<-EOF > console.sh
+         #! /bin/bash
+         stty rows ${rows} columns ${columns}
+         cd "\${virtme_chdir}"
+         HOME=${HOME}
+         fish  # use use zsh, tmux, byobu, screen, etc.
```

Comment:
I was wondering if we should provide a helper script in `virtme/guest`, so when `--vsock` is specified without any argument we can use this default shell wrapper. Something like `virtme/guest/virtme-vsock-console`.

And maybe this wrapper should look like the following:

```
#!/bin/bash

read -r rows columns <<< "$(stty size)"
stty rows ${rows} columns ${columns}
HOME=$(getent passwd ${virtme_user} | cut -d: -f6)
cd ${virtme_chdir}
stty echo
exec su - ${virtme_user}
```

What do you think?

Reply:
I think that would be a good idea.

I didn't try to do this for two main reasons:
- We need to make sure the script is mounted in the VM. I guess we can always add another mount point, or use the guest one (but the user might not have the right to modify it)
- We need a way to pass info about the terminal size (`stty size`) when `vng --vsock-connect` is launched: so after the VM has been started, and from a different terminal. For SSH, they have a control channel to pass all that, not here.

It is not impossible: I guess we could have a dedicated mount point, using a temp file on the host that would be populated when launching `vng --vsock-connect`. WDYT?

Should we eventually already merge this while we are discussing this, and improve this later in another PR?

About the wrapper:

> `read -r rows columns <<< "$(stty size)"`

This needs to be done from the host. Inside, I guess stty size will always return 80x24 (or the one of the terminal which launched the VM?), no?

> `HOME=$(getent passwd ${virtme_user} | cut -d: -f6)`

Good idea, but we should use: `${virtme_user:-root}` if root is used I suppose. Same later.

> `stty echo`

Nice, it looks like it is fixing the issue I had with `bash`! Is it why you added it?

> `exec su - ${virtme_user}`

We need to do the `cd ${virtme_chdir}` after that one. (And also use `${virtme_user:-root}`)

Reply:
> I think that would be a good idea.
> 
> I didn't try to do this for two main reasons:
> 
> * We need to make sure the script is mounted in the VM. I guess we can always add another mount point, or use the guest one (but the user might not have the right to modify it)

If we put the script in the guest_tools path they'll always be available (see `virtme/guest`, it's the same dir where the init script/binary is also placed), so no need to add another mount point.

> * We need a way to pass info about the terminal size (`stty size`) when `vng --vsock-connect` is launched: so after the VM has been started, and from a different terminal. For SSH, they have a control channel to pass all that, not here.

Ah good point about the terminal size. I'm not sure how to fix this, I have to think about it.

> 
> It is not impossible: I guess we could have a dedicated mount point, using a temp file on the host that would be populated when launching `vng --vsock-connect`. WDYT?

Honestly I'd prefer to not add a dedicated mount point for this, in theory we already have a lot of shared dirs between host and guest. Or maybe we can introduce a shared rw mount point to exchange info between host and guest... and use it also for other purposes, something similar to the guest_tools path, for example guest_data? We can probably introduce this later and ignore the terminal size issue for now...

> 
> Should we eventually already merge this while we are discussing this, and improve this later in another PR?

Yes, sounds good to me. We can definitely merge as it and improve it later in-tree.

> 
> About the wrapper:
> 
> > `read -r rows columns <<< "$(stty size)"`
> 
> This needs to be done from the host. Inside, I guess stty size will always return 80x24 (or the one of the terminal which launched the VM?), no?

Yep, you're right.

> 
> > `HOME=$(getent passwd ${virtme_user} | cut -d: -f6)`
> 
> Good idea, but we should use: `${virtme_user:-root}` if root is used I suppose. Same later.

Correct.

> 
> > `stty echo`
> 
> Nice, it looks like it is fixing the issue I had with `bash`! Is it why you added it?

Yep, this fixed bash for me.

> 
> > `exec su - ${virtme_user}`
> 
> We need to do the `cd ${virtme_chdir}` after that one. (And also use `${virtme_user:-root}`)

Correct, and maybe it's fine to just switch to the user's $HOME. At the end it'd be like a separate login (similar to ssh), so we don't have to honor `${virtme_chdir}`.

Reply:
> > I think that would be a good idea.
> > I didn't try to do this for two main reasons:
> > 
> > * We need to make sure the script is mounted in the VM. I guess we can always add another mount point, or use the guest one (but the user might not have the right to modify it)
> 
> If we put the script in the guest_tools path they'll always be available (see `virtme/guest`, it's the same dir where the init script/binary is also placed), so no need to add another mount point.

Indeed, but if we need to modify this script on the host (for the terminal size), that will be an issue :-/

> > It is not impossible: I guess we could have a dedicated mount point, using a temp file on the host that would be populated when launching `vng --vsock-connect`. WDYT?
> 
> Honestly I'd prefer to not add a dedicated mount point for this, in theory we already have a lot of shared dirs between host and guest. Or maybe we can introduce a shared rw mount point to exchange info between host and guest... and use it also for other purposes, something similar to the guest_tools path, for example guest_data? We can probably introduce this later and ignore the terminal size issue for now...

Yes, that's a good idea. But we would need a way to avoid conflicts when multiple VMs are started in parallel, and a way to find this directory when `vng --vsock-connect` is launched.

During a break, I started to look for a solution. I can commit that. It introduces a new mount point if root is not mounted. Maybe easier to discuss that around the code.

Reply:
I just opened #198, just in case you still want to merge the first part now :)